state_test.go

```diff
@@ -1680,15 +1686,40 @@ func TestMemberlist_Gossip(t *testing.T) {
 	a3 := alive{Node: "172.0.0.1", Addr: []byte{172, 0, 0, 1}, Incarnation: 1}
 	m1.aliveNode(&a3, nil, false)
 
-	// Gossip should send all this to m2
-	m1.gossip()
+	// Gossip should send all this to m2. Retry a few times because it's UDP and
+	// timing and stuff makes this flaky without.
+	retry(t, 5, 10*time.Millisecond, func(failf func(string, ...interface{})) {
+		m1.gossip()
```

Comment:
Should this be outside of the retry loop? Seems like we should be retrying to wait for the arrival of delayed messages and not to also broadcast more times. (Same comment for L1735)